### PR TITLE
feat(slider) Add new slider component

### PR DESCRIPTION
**Pull request by vuluongj20**
**Before ——**
image

**After ——**
image
image
image
image

### CODE REVIEW

File: static/app/components/slider/index.tsx [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+   * (This prop is now deprecated - slider ranges need to have consistent, evenly
+   * spaced values. Use `min`/`max`/`step` instead.)
+   *
+   * Custom array of selectable values on the track. If specified, the `min`/`max`/`step`
+   * props will be ignored. Make sure the array is sorted.
+   * @deprecated
+   */
+  allowedValues?: number[];
+  className?: string;
+  disabled?: boolean;
```
Comment (vuluongj20) [inline static/app/components/slider/index.tsx:24]: Disabled state:
image

File: static/app/components/slider/index.tsx [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+  error?: boolean;
```
Comment (vuluongj20) [inline static/app/components/slider/index.tsx:25]: Error state:
image

File: static/app/components/slider/index.tsx [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+              if (!node) {
+                return;
+              }
+
+              refs.current = [
+                ...refs.current.slice(0, index),
+                node,
+                ...refs.current.slice(index + 1),
+              ];
+            }}
```
Comment (vuluongj20) [inline static/app/components/slider/index.tsx:266]: Build a list of refs to each thumb's input element. 

**If there is only one thumb**, then `Slider`'s ref will point to the input element for just that thumb
image
```js 
{current: <input … />})
```

**If there are two or more thumbs**, then `Slider`'s ref will point to an array of input elements, each corresponding to a thumb.
image
```js 
{current: [<input … />, <input … />]})
```

File: static/app/components/slider/thumb.tsx [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    >
+      {showLabel && (
+        <SliderThumbLabel
+          aria-hidden
+          style={{
+            // Align thumb label with the track's edges. At 0% (min value) the label's
+            // leading edge should align with the track's leading edge. At 100% (max value)
+            // the label's trailing edge should align with the track's trailing edge.
+            left: `${state.getThumbPercent(index ?? 0) * 100}%`,
+            transform: `translateX(${-state.getThumbPercent(index ?? 0) * 100}%)`,
```
Comment (vuluongj20) [inline static/app/components/slider/thumb.tsx:50]: image
<br />
image

**Comment by evanpurkhiser** [comment]
Looks great! What was wrong with the old one? Not flexible enough?

**Comment by vuluongj20** [comment]
@evanpurkhiser the old one doesn't support tick marks/labels, which we need for starfish settings

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**